LayoutKit/Layouts/SizeLayout.swift

```diff
@@ -88,28 +151,90 @@ public class SizeLayout<V: View>: BaseLayout<V>, Layout {
                   config: config)
     }
 
-    public func measurement(within maxSize: CGSize) -> LayoutMeasurement {
-        let size = CGSize(width: width ?? .max, height: height ?? .max)
-        var constrainedSize = size.sizeDecreasedToSize(maxSize)
-
-        // If at least one dimension is nil, then we need to measure the sublayout to inherit its value (zero if there is no sublayout).
-        if width == nil || height == nil {
-            let subsize = sublayout?.measurement(within: constrainedSize).size ?? CGSizeZero
-            if width == nil {
-                constrainedSize.width = subsize.width
-            }
-            if height == nil {
-                constrainedSize.height = subsize.height
-            }
+    public convenience init(maxSize: CGSize,
```

Comment:
I experimented with this approach and it just made the callsites a little more verbose, so going to leave it as it is for now.

```
SizeLayout(width: .equals(45), height: .lessThanOrEqual(50))
SizeLayout(width: 45, maxHeight: 50)
```

I promoted SizeLayout(width:height:) to a designated initializer for backwards compatibility with current subclasses.